src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```
@@ -2,21 +2,42 @@
 
 import java.util.Optional;
 import mate.academy.hibernate.relations.dao.CountryDao;
+import mate.academy.hibernate.relations.exception.DataProcessingException;
 import mate.academy.hibernate.relations.model.Country;
-import org.hibernate.SessionFactory;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
 
 public class CountryDaoImpl extends AbstractDao implements CountryDao {
-    public CountryDaoImpl(SessionFactory sessionFactory) {
+    public CountryDaoImpl(org.hibernate.SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
     @Override
     public Country add(Country country) {
-        return null;
+        Transaction tx = null;
+        try (Session session = factory.openSession()) {
+            tx = session.beginTransaction();
+            session.save(country);
+            tx.commit();
+            return country;
+        } catch (Exception e) {
+            if (tx != null) {
+                try {
+                    tx.rollback();
+                } catch (Exception ignored) {
```

Swallowing exceptions here is not a good practice. If the rollback operation fails, it might indicate a more serious problem with the database connection or transaction manager. It's better to log this exception or wrap it in a `DataProcessingException`, similar to the implementation in `ActorDaoImpl`.